sim/firesim-lib/src/main/scala/bridges/TerminationBridge.scala

```diff
@@ -0,0 +1,136 @@
+  val validH = InputChannel(targetIO.valid)
+}
+
+class TerminationBridge(params: TerminationBridgeParams) extends BlackBox
+    with Bridge[TerminationBridgeHostIO, TerminationBridgeModule] {
+  val io = IO(new TerminationBridgeTargetIO(params))
+  val bridgeIO = new TerminationBridgeHostIO(params)(io)
+
+  val constructorArg = Some(params)
+
+  generateAnnotations()
+}
+
+object TerminationBridge {
+
+  private def annotateTerminationBridge(clock: Clock, lReset: Reset, finish: Seq[Bool], errCode: Seq[TerminationCondition], params: TerminationBridgeParams): TerminationBridge = {
+    val finishAll = finish.reduce(_||_)
+    val errMessageID = PriorityEncoder(finish)
+    //val gReset = WireInit(false.B)
+    //GlobalResetCondition.sink(gReset)
+    val terminationBridgeTarget = Module(new TerminationBridge(params))
+    terminationBridgeTarget.io.errCode := errMessageID
+    terminationBridgeTarget.io.valid := finishAll && !lReset.asBool
+    terminationBridgeTarget.io.clock := clock 
+    terminationBridgeTarget  
+  }
+
+  def apply(clock: Clock, lReset: Reset, finish: Seq[Bool], errCode: Seq[TerminationCondition], params: TerminationBridgeParams): TerminationBridge = {
+    annotateTerminationBridge(clock, lReset, finish, errCode, params)
+  }
+
+  def apply(clock: Clock, lReset: Reset, finish: Bool, errCode: UInt, params: TerminationBridgeParams): TerminationBridge = {
+    //val gReset = WireInit(false.B)
+    //GlobalResetCondition.sink(gReset)
+    val terminationBridgeTarget = Module(new TerminationBridge(params))
+    terminationBridgeTarget.io.errCode := errCode
+    terminationBridgeTarget.io.valid := finish && !lReset.asBool
+    terminationBridgeTarget.io.clock := clock 
+    terminationBridgeTarget  
+  }
+
+  def apply(finish: Seq[Bool], errCode: Seq[TerminationCondition], params: TerminationBridgeParams): TerminationBridge = {
+    annotateTerminationBridge(Module.clock, Module.reset, finish, errCode, params)
+  }
+
+}
+
+class TerminationBridgeModule(params: TerminationBridgeParams)(implicit p: Parameters) extends BridgeModule[TerminationBridgeHostIO]()(p) {
+  lazy val module = new BridgeModuleImp(this) {
+    // This creates the interfaces for all of the host-side transport
+    // AXI4-lite for the simulation control bus, =
+    // AXI4 for DMA
+    val io = IO(new WidgetIO())
+
+
+
+    // This creates the host-side interface of your TargetIO
+    val hPort = IO(new TerminationBridgeHostIO(params)())
+    
+    //Resume the simulation after the target called for termination
+    val resume = WireInit(false.B)
+
+    val statusDone = Module(new Queue(Bool(), 1))
+    val errCode = Module(new Queue(hPort.errCodeH.bits.cloneType, 1))
+
+    val cycleCountWidth = 16 //Data width is capped by nastikey 
+
+    val tokenCounter = RegInit(0.U(cycleCountWidth.W))
+
+
+    statusDone.io.enq <> hPort.validH
+    errCode.io.enq <> hPort.errCodeH 
+
+    val terminateResume = !statusDone.io.deq.bits || resume
```

Comment:
We decided to strip out the resume right?

Reply:
I thought it will be useful to the user. Or is it only for testing purpose?

Reply:
Done